File: app/src/main/java/dev/ragnarok/fenrir/db/model/entity/feedback/NewCommentEntity.java
package dev.ragnarok.fenrir.db.model.entity.feedback;

import com.google.gson.annotations.SerializedName;

import dev.ragnarok.fenrir.db.model.entity.CommentEntity;
import dev.ragnarok.fenrir.db.model.entity.Entity;
import dev.ragnarok.fenrir.db.model.entity.EntityWrapper;
import dev.ragnarok.fenrir.model.feedback.FeedbackType;

public class NewCommentEntity extends FeedbackEntity {

    @SerializedName("commented")
    private EntityWrapper commented = new EntityWrapper();
    @SerializedName("comment")
    private CommentEntity comment;

    @SuppressWarnings("unused")
    public NewCommentEntity() {
    }

    public NewCommentEntity(@FeedbackType int type) {
        setType(type);
    }

    public CommentEntity getComment() {
        return comment;
    }

    public NewCommentEntity setComment(CommentEntity comment) {
        this.comment = comment;
        return this;
    }

    public Entity getCommented() {
        return commented.get();
    }

    public NewCommentEntity setCommented(Entity commented) {
        this.commented = new EntityWrapper().wrap(commented);
        return this;
    }
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/db/model/entity/feedback/NewCommentEntity.kt
package dev.ragnarok.fenrir.db.model.entity.feedback

import androidx.annotation.Keep
import dev.ragnarok.fenrir.db.model.entity.CommentEntity
import dev.ragnarok.fenrir.db.model.entity.DboEntity
import dev.ragnarok.fenrir.model.feedback.FeedbackType
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Keep
@Serializable
@SerialName("new_comment")
class NewCommentEntity : FeedbackEntity {
    private var commented: DboEntity? = null

    var comment: CommentEntity? = null
        private set

    @Suppress("UNUSED")
    constructor()
    constructor(@FeedbackType type: Int) {
        this.type = type
    }

    fun setComment(comment: CommentEntity?): NewCommentEntity {
        this.comment = comment
        return this
    }

    fun getCommented(): DboEntity? {
        return commented
    }

    fun setCommented(commented: DboEntity?): NewCommentEntity {
        this.commented = commented
        return this
    }
}